src/rust/cryptography-x509/src/certificate.rs

```diff
@@ -13,6 +15,46 @@ pub struct Certificate<'a> {
     pub signature: asn1::BitString<'a>,
 }
 
+impl Certificate<'_> {
+    /// Checks whether this certificate contains any duplicated extensions;
+    /// returns the first duplicate's OID.
+    ///
+    /// If no duplicates are found, returns `None`.
+    pub fn check_duplicate_extensions(&self) -> Option<asn1::ObjectIdentifier> {
```

Comment:
Here's the new helper; I removed the corresponding code in the Python bindings and inserted a call to it.

(I haven't inserted the same call inside of `get_extension`, since I figure it should happen in the callsite(s) to `get_extension`).

Reply:
(I also broke it, debugging now.)

Reply:
Oh, this is my own fault -- I thought `extensions` was the only place this gets built up, but there are a few others. So this duplication logic needs to be reusable from contexts that don't have a full `Certificate`.